packages/perseus/src/util/graphie.ts

```diff
@@ -580,93 +614,324 @@ export class Graphie {
         $.extend(this.currentStyle, processed);
     }
 
-    grid(
-        xr: Interval,
-        yr: Interval,
-        styleAttributes?: Record<string, any>,
-    ): RaphaelSet {
-        throw new Error("grid called on uninitialized Graphie");
-    }
+    grid(xr: Interval, yr: Interval, style?: Record<string, any>): RaphaelSet {
+        return this.withStyle(style, () => {
+            const step: any = this.currentStyle.step || [1, 1];
+            const set = this.raphael.set();
 
-    // circle is a stub that gets overwritten with a function from drawingTools
-    // in createGraphie
-    circle(
-        center: Coord,
-        radius: number,
-        style?: Record<string, any>,
-    ): RaphaelElement {}
-
-    // rect is a stub that gets overwritten with a function from drawingTools
-    // in createGraphie
-    rect(
-        left: number,
-        bottom: number,
-        width: number,
-        height: number,
-        style?: Record<string, any>,
-    ): RaphaelElement {}
+            let x = step[0] * Math.ceil(xr[0] / step[0]);
+            for (; x <= xr[1]; x += step[0]) {
+                set.push(this.line([x, yr[0]], [x, yr[1]]));
+            }
 
-    // ellipse is a stub that gets overwritten with a function from drawingTools
-    // in createGraphie
-    ellipse(
-        center: Coord,
-        radii: Coord,
-        style?: Record<string, any>,
-    ): RaphaelElement {}
+            let y = step[1] * Math.ceil(yr[0] / step[1]);
+            for (; y <= yr[1]; y += step[1]) {
+                set.push(this.line([xr[0], y], [xr[1], y]));
+            }
 
-    fixedEllipse(
-        center: Coord,
-        radii: Coord,
-        scale: number,
-        padding: number,
-    ): PositionedShape {
-        throw new Error("fixedEllipse called on uninitialized Graphie");
+            return set;
+        });
     }
 
     arc(
         center: Coord,
-        radii: Coord,
+        radius: Coord,
         startAngle: number,
         endAngle: number,
         sector: boolean,
         style?: Record<string, any>,
-    ): RaphaelElement {}
+    ): RaphaelElement {
+        return this.withStyle(style, () => {
+            startAngle = ((startAngle % 360) + 360) % 360;
+            endAngle = ((endAngle % 360) + 360) % 360;
+
+            const cent = this.scalePoint(center);
+            const radii = this.scaleVector(radius);
+            const startVector = polar(radius, startAngle);
+            const endVector = polar(radius, endAngle);
+
+            const startPoint = this.scalePoint([
+                +(center[0] + startVector[0]).toFixed(6),
+                +(center[1] + startVector[1]).toFixed(6),
+            ]);
+            const endPoint = this.scalePoint([
+                +(center[0] + endVector[0]).toFixed(6),
+                +(center[1] + endVector[1]).toFixed(6),
```

Comment:
I should figure out a better way of rounding these, or a better way of phrasing the test so we don't have to round them.